Source/WebCore/editing/markup.cpp

```diff
@@ -1458,11 +1458,11 @@ static inline bool hasOneTextChild(ContainerNode& node)
 
 static inline bool hasMutationEventListeners(const Document& document)
 {
-    return document.hasListenerType(Document::DOMSUBTREEMODIFIED_LISTENER)
-        || document.hasListenerType(Document::DOMNODEINSERTED_LISTENER)
-        || document.hasListenerType(Document::DOMNODEREMOVED_LISTENER)
-        || document.hasListenerType(Document::DOMNODEREMOVEDFROMDOCUMENT_LISTENER)
-        || document.hasListenerType(Document::DOMCHARACTERDATAMODIFIED_LISTENER);
+    return document.hasListenerType(Document::ListenerType::DOMSubtreeModified)
```

Comment:
Now that we use an OptionSet, it may be nice to leverage OptionSet::containsAny(). Maybe it would look something like:
```
return document.hasAnyListenerOfType({ Document::ListenerType::DOMSubtreeModified, ListenerType::DOMNodeInserted, ... });
```

Just a suggestion. You don't have to do it in this patch.